src/lib/y2partitioner/widgets/disk_table.rb

```diff
@@ -22,13 +22,15 @@ def initialize(disks, pager)
 
       # table items. See CWM::Table#items
       def items
+        probed_graph = Y2Storage::StorageManager.instance.y2storage_probed
```

Comment:
This means that every single widget assumes that we are always working in the "probed" devicegraph. That kind of coupling is quite bad for reusability, testability, etc.

I know I'm repeating myself a lot, but a good Partitioner class should accept an initial devicegraph -it doesn't matter whether it's `Y2Storage::StorageManager.instance.probed` or another one programmatically created for testing (let's say via a factory)- and should return another devicegraph with the result.

Of course, the YaST `partitioner` client will run that partitioner with the "probed" devicegraph and will write the result to the "staging" one.

But every single piece of the partitioner should not be aware of those two global devicegraphs and should not assume we are always operating on them. That's pretty much the infamous targetmap approach.